packages/odata/src/services/grid-odata.service.ts

```diff
@@ -426,8 +426,20 @@ export class GridOdataService implements BackendService {
             fieldName = titleCase(getHtmlStringOutput(fieldName || ''));
           }
 
-          // StartsWith + EndsWith combo
-          if (operator === OperatorType.startsWithEndsWith && Array.isArray(searchTerms) && searchTerms.length === 2) {
+          let predicateResult: string | undefined = undefined;
+          if (typeof this._odataService?.options?.filterPredicate === 'function') {
```

Comment:
I took a quick look at `this._odataService` and since it's instantiated in the constructor, I believe we can safely assume that it's always filled and we should remove optional chaining from it, not just yours but everywhere else in this file

```diff
- this._odataService?.options
+ this._odataService.options
```

Reply:
PRs with you feel like talking to a Linter ... no excuses even for DraftPRs 🤣